netket/graph/_lattice_edge_logic.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2022 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import numpy as np
+from scipy.spatial import cKDTree
+from scipy.sparse import find, triu
+from netket.utils.float import comparable, is_approx_int
+
+from typing import Tuple
+
+# Helper functions
+
+
+def create_site_positions(basis_vectors, extent, site_offsets, extra_shells=None):
+    """Generates the coordinates of all lattice sites.
+
+    extra_shells, if used, must be a vector of the same length as extent"""
```

Comment:
Can you write a better docstring? what does `extra_shells` do? it seems to me the defeault is `np.zeros(extent.size, dtype=int)`...